test/test_access_om2_config.py

```diff
@@ -0,0 +1,116 @@
+from datetime import timedelta
+import re
+
+import pytest
+import f90nml
+
+from util import get_git_branch_name
+
+# Mutually exclusive topic keywords
+TOPIC_KEYWORDS = {
+    'spatial extent': {'global', 'regional'},
+    'forcing product':	{'JRA55', 'ERA5'},
+    'forcing mode': {'repeat-year', 'ryf', 'repeat-decade', 'rdf',
+                        'interannual', 'iaf'},
+    'model': {'access-om2', 'access-om2-025', 'access-om2-01'}
+}
+
+
+class AccessOM2Branch:
+    """Use the naming patterns of the branch name to infer informatiom of
+    the ACCESS-OM2 config"""
+
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        self.set_resolution()
+
+        self.is_high_resolution = self.resolution in ['025deg', '01deg']
+        self.is_bgc = 'bgc' in branch_name
+
+    def set_resolution(self):
+        # Resolutions are ordered, so the start of the list are matched first
+        resolutions = ['025deg', '01deg', '1deg']
+        self.resolution = None
+        for res in resolutions:
+            if res in self.branch_name:
+                self.resolution = res
+                return
+        # TODO Should unknown resolutions fail the pytests or be ignored?
```

Comment:
I think fail, so they can be flagged and explicit support added.

Reply:
Done